tests/test_utils.py

```diff
@@ -0,0 +1,247 @@
+
+
+def delete_project_if_it_exists(project_name):
+    """"""
+    if os.path.isdir(
+        os.path.join(appdirs.user_data_dir("DataShuttle"), project_name)
+    ):
+        shutil.rmtree(
+            os.path.join(
+                appdirs.user_data_dir("DataShuttle"),
+                project_name,
+            )
+        )
+
+
+def get_test_config_arguments_dict(
+    set_as_defaults=None, required_arguments_only=None
+):
+    """
+    Retrieve configs, either the required configs
+    (for project.make_config_file()), all configs (default)
+    or non-default configs. Note that default configs here
+    are the expected default arguments in project.make_config_file().
+    """
+    dict_ = {
+        "local_path": r"Not:/a/real/local/directory",
+        "ssh_to_remote": False,
+        "remote_path_local": r"/Not/a/real/remote_local/directory",
+        "remote_path_ssh": r"/not/a/real/remote_ssh/directory",
+    }
+
+    if required_arguments_only:
+        return dict_
+
+    if set_as_defaults:
+        dict_.update(
+            {
+                "remote_host_id": None,
+                "remote_host_username": None,
+                "sub_prefix": "sub-",
+                "ses_prefix": "ses-",
+                "use_ephys": True,
+                "use_ephys_behav": True,
+                "use_ephys_behav_camera": True,
+                "use_behav": True,
+                "use_behav_camera": True,
+                "use_histology": True,
+                "use_imaging": True,
+            }
+        )
+    else:
+        dict_.update(
+            {
+                "ssh_to_remote": True,
+                "remote_host_id": "test_remote_host_id",
+                "remote_host_username": "test_remote_host_username",
+                "sub_prefix": "testsub-",
+                "ses_prefix": "testses-",
+                "use_ephys": False,
+                "use_ephys_behav": False,
+                "use_ephys_behav_camera": False,
+                "use_behav": False,
+                "use_behav_camera": False,
+                "use_histology": False,
+                "use_imaging": False,
+            }
+        )
+
+    return dict_
+
+
+# ----------------------------------------------------------------------------------------------------------
+# Test Helpers
+# ----------------------------------------------------------------------------------------------------------
+
+
+def check_directory_tree_is_correct(
+    project, base_dir, subs, sessions, directory_used
+):
+    """
+    Automated test that directories are made based
+    on the  structure specified on project itself.
+
+    Cycle through all experiment type (defined in
+    project._ses_dirs()), sub, sessions and check that
+    the expected file exists. For  subdirs, recursively
+    check all exist.
+
+    Directories in which directory_used[key] (where key
+    is the cannoincal dict key in project._ses_dirs())
+    is not used are expected  not to be made, and this
```

Comment:
there is a typo here. Also I don't really understand the sentence in this paragraph